### ISSUE TITLE
Animation only shows first frame when falling after jump

### ISSUE THREAD

**Issue by DanielSmith1239** [4ian/GDevelop#863]
## Describe the bug
I'm doing an "Is Falling" check on my player to set a falling animation. When the player falls off of a platform (without jumping), it displays the animation properly. However, when the player begins falling after a jump, only the first frame of the animation is shown. 

## To Reproduce
1. Create a player object with a falling animation.
2. Add a falling trigger and set the animation when the trigger is hit.
* If applicable, add screenshots to help explain your problem.

## Other details
* Game link: http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1547160570545
* I'm using the editor on my Mac.

**Comment by 4ian** [comment]
My guess is that the condition jumping and falling are BOTH true at the same time, so you're changing twice the animation, so frame is reset to 0 every time.
To fix this, ensure that only one change of animation is made at the same time (by using a negated condition for example or something else)

**Comment by DanielSmith1239** [comment]
How should I properly change the animations then?

**Comment by zatsme** [comment]
Just add Trtigger Once to both conditions 👍 🥇

**Comment by 4ian** [comment]
@DanielSmith1239 Can you confirm if the solution given by zatsme is working? We could maybe include this in an example.

**Comment by zatsme** [comment]
I modified the platformer example with dummy sprites to see it working.  It would be good to add this to the platformer example but the main character does not have the required animations.  Maybe a new platformer example with new sprites could be made?

**Comment by 4ian** [comment]
Yeah, either that, or see if the sprites from the Kenney pack have an animation for falling, or use a slightly different image :)

**Comment by blurymind** [comment]
One thing I tend to do in other engines to avoid problems like this is to
create a utility function for switching to other animations. It takes the
new animation name as first parameter and compares it to the current
animation name. If they are different, then set the object to the new
animation. Of they aren't don't do it. Gdevelop has the trigger once
action, which should sort this

**Comment by zatsme** [comment]
Trigger once works as it doesn't trigger again until falling is over, I'll have a look at changing the platformer example if you like 🙂.  Lots of people just don't understand the event system, I think we need more in the wiki, specifically tutorials, to make it obvious.

**Comment by Lizard-13** [comment]
@blurymind GD already does that, ignores animation switching if it's the same, but people switch between multiple animations in the same frame, reseting it.

Yeah, it has to be more documented indeed, an explanation about animation switching/reset in the sprite object, and a reminder in the platformer article with the right way to do it.

**Comment by DanielSmith1239** [comment]
Sorry for the late response, I'll be able to try out @zatsme's solution a bit later today.

**Comment by 4ian** [comment]
Sweet, glad it's working! I think I'll start a series of "how to" page on the wiki to address such questions and give some visibility/searchability to the answer :)
